spec/lib/board_spec.rb

```diff
@@ -0,0 +1,252 @@
+      board = Board.new(3)
+      expect(board.translate_move(1)).to eql([0,0])
+      expect(board.translate_move(4)).to eql([1,0])
+    end
+  end
+
+  describe "#make_move" do
+    it "#should return true if move is valid, return false if not" do
+      board = Board.new(3)
+      expect(board.make_move('3',GameConstants::PLAYER_ONE_MARK)).to be true
+      expect(board.make_move('10',GameConstants::PLAYER_TWO_MARK)).to be false
+    end
+  end
+
+  describe "#update_by_vertically" do
+    it "should change state when finished by vertically" do
+      board = Board.new(3)
+      board.make_move('1', GameConstants::PLAYER_ONE_MARK)
+      board.make_move('4', GameConstants::PLAYER_ONE_MARK)
+      board.make_move('7', GameConstants::PLAYER_ONE_MARK)
+      expect(board.game_state).to eql(GameConstants::PLAYER_ONE_WON)
+
+      board.reset
+
+      board.make_move('2', GameConstants::PLAYER_ONE_MARK)
+      board.make_move('5', GameConstants::PLAYER_ONE_MARK)
+      board.make_move('8', GameConstants::PLAYER_ONE_MARK)
+      expect(board.game_state).to eql(GameConstants::PLAYER_ONE_WON)
+
+      board.reset
+
+      board.make_move('3', GameConstants::PLAYER_ONE_MARK)
+      board.make_move('6', GameConstants::PLAYER_ONE_MARK)
+      board.make_move('9', GameConstants::PLAYER_ONE_MARK)
+      expect(board.game_state).to eql(GameConstants::PLAYER_ONE_WON)
+
+      board.reset
+
+      board.make_move('1', GameConstants::PLAYER_TWO_MARK)
+      board.make_move('4', GameConstants::PLAYER_TWO_MARK)
+      board.make_move('7', GameConstants::PLAYER_TWO_MARK)
+      expect(board.game_state).to eql(GameConstants::PLAYER_TWO_WON)
+
+      board.reset
+
+      board.make_move('2', GameConstants::PLAYER_TWO_MARK)
+      board.make_move('5', GameConstants::PLAYER_TWO_MARK)
+      board.make_move('8', GameConstants::PLAYER_TWO_MARK)
+      expect(board.game_state).to eql(GameConstants::PLAYER_TWO_WON)
+
+      board.reset
+
+      board.make_move('3', GameConstants::PLAYER_TWO_MARK)
+      board.make_move('6', GameConstants::PLAYER_TWO_MARK)
+      board.make_move('9', GameConstants::PLAYER_TWO_MARK)
+      expect(board.game_state).to eql(GameConstants::PLAYER_TWO_WON)
+
+      board = Board.new(5)
+      board.make_move('1', GameConstants::PLAYER_ONE_MARK)
+      board.make_move('6', GameConstants::PLAYER_ONE_MARK)
+      board.make_move('11', GameConstants::PLAYER_ONE_MARK)
+      board.make_move('16', GameConstants::PLAYER_ONE_MARK)
+      board.make_move('21', GameConstants::PLAYER_ONE_MARK)
+      expect(board.game_state).to eql(GameConstants::PLAYER_ONE_WON)
+
+      board = Board.new(4)
+      board.make_move('1', GameConstants::PLAYER_TWO_MARK)
+      board.make_move('5', GameConstants::PLAYER_TWO_MARK)
+      board.make_move('9', GameConstants::PLAYER_TWO_MARK)
+      board.make_move('13', GameConstants::PLAYER_TWO_MARK)
+      expect(board.game_state).to eql(GameConstants::PLAYER_TWO_WON)
+    end
```

Comment:
Why not writing `board.make_move` inside a loop and generating the combinations dynamically. Rewriting this if needed would be painful

Reply:
we tried to change it with a loop but we both agreed to use basically make_move is better. 

` 3.times do
        board.make_move(counter.to_s, GameConstants::PLAYER_ONE_MARK)
        counter += 3
      end
`